hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -126,18 +128,27 @@ public DeletedBlockLogImpl(ConfigurationSource conf,
   }
 
   @Override
-  public List<DeletedBlocksTransaction> getFailedTransactions()
+  public List<DeletedBlocksTransaction> getFailedTransactions(int count)
       throws IOException {
     lock.lock();
     try {
       final List<DeletedBlocksTransaction> failedTXs = Lists.newArrayList();
       try (TableIterator<Long,
           ? extends Table.KeyValue<Long, DeletedBlocksTransaction>> iter =
                deletedBlockLogStateManager.getReadOnlyIterator()) {
-        while (iter.hasNext()) {
-          DeletedBlocksTransaction delTX = iter.next().getValue();
-          if (delTX.getCount() == -1) {
-            failedTXs.add(delTX);
+        if (count <= 0) {
+          while (iter.hasNext()) {
+            DeletedBlocksTransaction delTX = iter.next().getValue();
+            if (delTX.getCount() == -1) {
+              failedTXs.add(delTX);
+            }
+          }
+        } else {
+          while (iter.hasNext() && failedTXs.size() < count) {
+            DeletedBlocksTransaction delTX = iter.next().getValue();
+            if (delTX.getCount() == -1) {
```

Comment:
Should this be switched as well?
```suggestion
            if (delTX.getCount() == LIST_ALL_FAILED_TRANSACTIONS) {
```

Reply:
Oops, this is the special value of DeletedBlocksTransaction, not for the clear instant description.       
See `DeletedBlocksTransaction.Builder builder = block.toBuilder().setCount(-1);`